201 update dependencies

sqr00t opened the pull request:
- Removes awscli from requirements_dev.txt
- Updates following dependencies:
  - numpy 1.22.4 -> 1.24.4
  - scipy 1.8.1 -> 1.10.1
  - scikit-learn 0.23.2 -> 1.3.1
- Adapts @Aiden-RC 's snippet for download() function using boto3
- Adds pip dependency caching for "Unit Tests" workflow
- Adds windows-latest support to "Unit Tests" Actions workflow
- Utilise Path objects in setup.py
- Fixes cross-platform version tagging bug by:
  - Adding setuptools_scm to setup.py
  - Build config to a pyproject.toml file
- Removes manually downloading base SpaCy model

---

Thanks for contributing to Nesta's Skills Extractor Library 🙏!

If you have suggested changes to _code_ anywhere outside of the ExtractSkills class, please consult the checklist below.

Checklist ✔️🐍:

- [x] I have refactored my code out from `notebooks/` (no notebook code involved)
- [x] I have checked the code runs
- [x] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [x] Appropriate information has been added to `README`s
- [x] I have explained the feature in this PR or (better) in `output/reports/`
- [x] I have requested a code review

If you have suggested changes to _documentation_ (and/or the ExtractSkills class), please **ALSO** consult the checklist below.

Documentation Checklist ✔️📚:

- [x] I have run `make html` in `docs`
- [x] I have manually reviewed the `docs/build/*.html` files locally to ensure they have formatted correctly
- [x] I have pushed both relevant files AND their corresponding `docs/build/*.html` files

Aiden-RC commented on ojd_daps_skills/getters/download_public_data.py:
```diff
@@ -14,5 +13,43 @@ def download():
+from zipfile import ZipFile
+
+
+def download():
```
i'm unsure if you've realised but you've defined and called both the aws cli command aswell as using boto to download the data?
You've also overwritten the same function

sqr00t replied:
Brilliant spot! I had a long day. Thank you 🤣

Aiden-RC reviewed (changes requested):
Unsure why download function has been overwritten / called twice?
Potential oversight?

Aiden-RC commented:
Have tested, works perfectly for me.
As was running on windows had to make a few changes to get it to run ( that i had made previously but had to do a few other changes also to make it run on windows correctly.
New PR made for this at https://github.com/nestauk/ojd_daps_skills/pull/204
contains this PR changes aswell as windows related changes and windows pytest actions.

I would recommend removing aws cli section of readme like i did on my PR if mine doesn't get merged

Aiden-RC commented on README.md:
````diff
@@ -35,10 +35,6 @@ You will also need to download [spaCy's](https://spacy.io/models/en) `en_core_we
 python -m spacy download en_core_web_sm
 ```
 
-### AWS CLI
````
This conflicts with latest changes on the 'dev' branch.

sqr00t commented:
I've had a deeper look at the version tagging implementation.
Automatic versioning now should work across Windows and Mac:
- Added using setuptools_scm for this feature
- Added some build config in a new pyproject.toml file

Further work that may be needed:
- Major/ Minor releases may need to have it's own branch created manually, as this automatically updates the fix/hotfix version. I'm having another look at [this](https://riggaroo.dev/using-github-actions-to-automate-our-release-process/)

india-kerle commented:
great - i was able to create a new env, install dependencies and use the libraries core functionality with no errors. Looks like all tests on ubuntu and windows passed as well.
